components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -368,10 +368,20 @@ private Map<String, Object> getUserClaimsInOIDCDialect(OAuthAuthzReqMessageConte
         if (isEmpty(userAttributes)) {
             if (isLocalUser(authzReqMessageContext)) {
                 if (log.isDebugEnabled()) {
-                    log.debug("User attributes not found in cache. Trying to retrieve attribute for local user: " +
-                            authzReqMessageContext.getAuthorizationReqDTO().getUser());
+                    log.debug("User attributes not found in cache. Trying to retrieve attribute from auth " +
+                            "context for local user: " + authzReqMessageContext.getAuthorizationReqDTO().getUser());
+                }
+                userAttributes = authzReqMessageContext.getAuthorizationReqDTO().getUser()
+                        .getUserAttributes();
+                if (isEmpty(userAttributes)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("User attributes not found in auth context. Trying to retrieve attribute for " +
+                                "local user: " + authzReqMessageContext.getAuthorizationReqDTO().getUser());
```

Comment:
`authzReqMessageContext.getAuthorizationReqDTO().getUser()` It may log the user or to string here. Make sure it logs the userId here by using user.userId and avoid username in ogs

Reply:
get userId and log that, in that case, you don't need to have log.isDebugEnabled() check. It will be a redundant operation since log.isDebugEnabled() check for log enabled.